cloud-functions/cf-dsm-file-scanner/dsm-gcp-setup/cf-deploy.sh

```diff
@@ -0,0 +1,342 @@
+#
+# CONTAINER_REGISTRY assumes that the Google Artifact Registry is going to be
+# used to house the images, but there's no strong requirement for this.
+# If an alternate registry is desired, the most straightforward path would
+# be to allow the `CONTAINER_REGISTRY` pattern to be overridden.
+#
+# CONTAINER_FQ_NAME -> Container Fully-Qualified Name
+##
+CONTAINER_NAME="${CONTAINER_NAME:-$SERVICE_NAME}"
+CONTAINER_VERSION="${CONTAINER_VERSION:-latest}"
+CONTAINER_REGISTRY="$CLOUDSDK_RUN_REGION-docker.pkg.dev/$PROJECT_ID/dss"
+CONTAINER_FQ_NAME="$CONTAINER_REGISTRY/$CONTAINER_NAME:$CONTAINER_VERSION"
+
+function gcloudw {
+  gcloud "$@" $OTHER_GCLOUD_FLAGS
+}
+
+function check-for-image {
+  echo $CONTAINER_FQ_NAME
+  gcloudw container images describe "$CONTAINER_FQ_NAME" &> /dev/null
+}
+
+function check-for-service-account {
+  gcloudw iam service-accounts describe "$SERVICE_ACCOUNT" &> /dev/null
+}
+
+function create-service-account {
+  gcloudw iam service-accounts create "$SERVICE_ACCOUNT" \
+    --display-name="gsc-$SERVICE_NAME" \
+    --description="Service Account for the $SERVICE_NAME service" 1>/dev/null
+
+}
+
+function access-secret {
+  ##
+  # Intentially not redirecting the output here as its going
+  # to be the value of the secret and we want to be able to
+  # pipe & manipulate the stdout
+  ##
+  gcloudw secrets versions access "$SECRET_VERSION" \
+    --project="$PROJECT_ID" \
+    --secret="$SECRET_ID"
+}
+
+function check-for-subscription-topic {
+  gcloudw pubsub topics describe "$LISTEN_TOPIC_NAME"  &> /dev/null
+}
+
+function check-for-result-topic {
+  gcloudw pubsub topics describe "$RESULT_TOPIC_NAME"  &> /dev/null
+}
+
+function add-subscriber-iam-binding {
+  gcloudw pubsub topics add-iam-policy-binding "$LISTEN_TOPIC_NAME" \
+    --member "serviceAccount:$SERVICE_ACCOUNT" \
+    --role "roles/pubsub.subscriber" 1> /dev/null
+}
+
+function add-result-publisher-iam-binding {
+  gcloudw pubsub topics add-iam-policy-binding "$RESULT_TOPIC_NAME" \
+    --member "serviceAccount:$SERVICE_ACCOUNT" \
+    --role "roles/pubsub.publisher" 1> /dev/null
+}
+
+function deploy-service {
+  ##
+  # Take a look at the performance once this image is deployed.
+  # The mix of ClamAV and Java were running into OOM errors with
+  # a fair degree of frequency when using 2Gi. I traced the bulk
+  # of the issue to the ConcurrentDatabaseReload config value
+  # in clamd.conf and disabled it, but haven't performed thorough
+  # testing with the reduced memory amount.
+  #
+  # Try reducing the 4Gi to 2Gi below and see if things are still happy.
+  # This system may also be able to get by with only a single CPU.
+  #
+  ##
+  gcloudw run deploy "$SERVICE_NAME" \
+    --service-account="$SERVICE_ACCOUNT" \
+    --region="$CLOUDSDK_RUN_REGION" \
+    --platform="managed" \
+    --cpu="1" \
+    --memory="4Gi" \
+    --image="$CONTAINER_FQ_NAME" \
+    --min-instances="1" \
+    --max-instances="3" \
+    --env-vars-file="env.yaml" \
```

Comment:
Where is `env.yaml`?
I notice it is also missing from cf-file-scanner, but I don't understand how either deployment works without it.